api/src/controllers/provision.ts

```diff
@@ -92,9 +92,23 @@ export const provisionCallbackHandler = async (
 };
 
 const updateProvisionedNamespaces = async (profile: ProjectProfile): Promise<void> => {
+  const { RequestModel } = dm;
+
   try {
     await updateProvisionedProfile(profile);
 
+    if (!profile.id) {
```

Comment:
We could perhaps have a util function that checks the query params validation (besides the existing one that checks for POST body structure based on Model required types) and serve 400 bad request consistently?